### DISCUSSION TITLE
get block range

### DISCUSSION THREAD

**Discussion by 47vigen** [remirror/remirror#1810] [Q&A] [answered]
how can I get the range of the current block?
same as `usePositioner` but return range

**Comment by alirezahmz** [comment]
I have same issue :(

**Comment by ocavue** [comment, marked as answer]
```ts
import {useEditorState} from '@remirror/react'

const state = useEditorState()
const {$from, $to} = state.selection
const range = $from.blockRange($to)
```

The docs for `blockRange`: https://prosemirror.net/docs/ref/#model.ResolvedPos.blockRange

**Reply by gaowujian** [reply]
any idea? how to do this? Appreciate for the help in advance. https://github.com/remirror/remirror/discussions/1954
